src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCIBuildPlanService.java

```diff
@@ -36,27 +37,37 @@
      * @return the default build plan
      */
     public String generateDefaultBuildPlan(ProgrammingExercise programmingExercise) {
-        Optional<String> projectTypeName;
-        if (programmingExercise.getProgrammingLanguage().equals(ProgrammingLanguage.JAVA)) {
-            projectTypeName = Optional.of("maven");
-        }
-        else {
-            projectTypeName = Optional.empty();
-        }
-        String[] resourcePath = projectTypeName
-                .map(name -> new String[] { "templates", "gitlabci", programmingExercise.getProgrammingLanguage().name().toLowerCase(), name,
-                        programmingExercise.hasSequentialTestRuns() ? "sequentialRuns" : "regularRuns", FILE_NAME })
-                .orElseGet(() -> new String[] { "templates", "gitlabci", programmingExercise.getProgrammingLanguage().name().toLowerCase(),
-                        programmingExercise.hasSequentialTestRuns() ? "sequentialRuns" : "regularRuns", FILE_NAME });
-
+        final Optional<String> projectTypeName = getProjectTypeName(programmingExercise);
+        final Path resourcePath = buildResourcePath(programmingExercise.getProgrammingLanguage(), projectTypeName, programmingExercise.hasSequentialTestRuns());
         final Resource resource = resourceLoaderService.getResource(resourcePath);
+
         try {
             return StreamUtils.copyToString(resource.getInputStream(), Charset.defaultCharset());
         }
         catch (IOException ex) {
             log.error("Error loading template GitLab CI build configuration", ex);
             throw new IllegalStateException("Error loading template GitLab CI build configuration", ex);
         }
+    }
+
+    private static Optional<String> getProjectTypeName(final ProgrammingExercise programmingExercise) {
+        if (programmingExercise.getProgrammingLanguage().equals(ProgrammingLanguage.JAVA)) {
```

Comment:
Probably it would be better if you used == to compare two enums. Both are correct, but == is additionally null safe.

Reply:
Good point about null-safety: I swapped the two sides, so that the `equals` gets called on the constant.